examples/Backtesting/custom_analytical.py

```diff
@@ -1,124 +1,110 @@
-## Example for full simulation loop using a custom analytical test function
+## Optimizing a Custom Black-Box Function
 
-# This example shows a simulation loop for a single target with a custom test function as lookup.
-# That is, we perform several Monte Carlo runs with several iterations.
-# In addition, we also store and display the results.
+# This example demonstrates how to optimize a custom black-box function:
+# * We create a black-box callable and define the corresponding optimization scope,
+# * set up optimization strategies,
+# * and compare the resulting trajectories.
 
-# This example assumes some basic familiarity with using BayBE and how to use BoTorch test
-# functions in discrete searchspaces.
-# For further details, we thus refer to
-# - [`campaign`](./../Basics/campaign.md) for a basic example on how to use BayBE and
-# - [here](./../Searchspaces/continuous_space_custom_function.md) for how to use a custom function.
-
-### Necessary imports for this example
+### Imports
 
 import os
 
-import matplotlib.pyplot as plt
 import numpy as np
 import seaborn as sns
 
 from baybe import Campaign
-from baybe.objectives import SingleTargetObjective
-from baybe.parameters import NumericalDiscreteParameter
-from baybe.recommenders import (
-    BotorchRecommender,
-    RandomRecommender,
-    TwoPhaseMetaRecommender,
-)
+from baybe.parameters.numerical import NumericalContinuousParameter
+from baybe.recommenders import RandomRecommender
 from baybe.searchspace import SearchSpace
-from baybe.simulation import simulate_scenarios
+from baybe.simulation import label_columns, simulate_scenarios
 from baybe.targets import NumericalTarget
+from baybe.utils.plotting import create_example_plots
 
-### Parameters for a full simulation loop
+### Settings
 
-# For the full simulation, we need to define some additional parameters.
-# These are the number of Monte Carlo runs and the number of experiments to be conducted per run.
-
-# The parameter `POINTS_PER_DIM` controls the number of points per dimension.
-# Note that the searchspace will have `POINTS_PER_DIM**DIMENSION` many points.
+# Before we start, let us collect a few general settings for the example:
 
 SMOKE_TEST = "SMOKE_TEST" in os.environ
 
-N_MC_ITERATIONS = 2 if SMOKE_TEST else 5
-N_DOE_ITERATIONS = 2 if SMOKE_TEST else 5
-DIMENSION = 4
-BOUNDS = [(-2, 2), (-2, 2), (-2, 2), (-2, 2)]
-POINTS_PER_DIM = 3 if SMOKE_TEST else 10
-
-### Defining the test function
+BATCH_SIZE = 1
+N_MC_ITERATIONS = 2 if SMOKE_TEST else 20
+N_DOE_ITERATIONS = 2 if SMOKE_TEST else 30
+DIMENSION = 1 if SMOKE_TEST else 10
```

Comment:
I'd go with a dimension of at least 2 or 3, even in the `SMOKE_TEST` since you speak about a "high-dimensional quadratic function" and 1 is NOT high-dimensional 😆

Reply:
Have changed it to 2 👍🏼 But note: not because we're claiming that the example is "high-dimensional". For the smoke test, this is irrelevant. It's only purpose is to test code execution, in the fastest possible way. The reason why I increased it because it then properly tests the lists case (i.e. otherwise, the test would silently pass in cases where there are hardcoded references to a single parameter object)